### PR TITLE
Upgrade app to 1.5

### PR DESCRIPTION
**Pull request by christinamegura**
## DESCRIPTION
This PR updates card mapping so that:
- `isFeatured` card display logic is not used on the home feed
- Default cards will include sermon series, sermons, worship, podcast, articles, pastor note, church updates, digital classes, stories, and Willow creative
- Highlight cards will include devotionals and study guides

### What does this PR do, or why is it needed?
This PR 
1. Upgrades the client and API to v1.5.0 in order to
2. Update both vertical and horizontal card mapping by adding an override for the `Component` prop of the `cardComponentMapper`

## TODO:

- [ ] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [tag-issues-here]
- [ ] No new warnings in tests, in storybook, and in-app
- [ ] Upload GIF(s) of iOS and Android
- [ ] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

File: packages/apollos-api/src/data/RockConstants.js [hunk 1]
```diff
@@ -1,20 +0,0 @@
...
-  async createOrFindInteractionComponent({
-    componentName,
-    channelId,
-    entityId,
-  }) {
-    return this.findOrCreate({
-      model: 'InteractionComponents',
-      objectAttributes: {
-        Name: componentName,
-        InteractionChannelId: channelId, // In willow, this field is InteractionChannel instead of Channel for unknown reasons.
```
Comment (vinnyjth) [inline packages/apollos-api/src/data/RockConstants.js:13]: Any reason you deleted this file?

File: packages/WillowCreekApp/src/index.js [hunk 2]
```diff
@@ -93,23 +91,16 @@ function getActiveRouteName(navigationState) {
...
             }}
-            onNavigationStateChange={(prevState, currentState) => {
-              const currentScreen = getActiveRouteName(currentState);
-              const prevScreen = getActiveRouteName(prevState);
-
-              if (prevScreen !== currentScreen) {
-                track({ eventName: `Viewed ${currentScreen}` });
-              }
-            }}
+            {...props}
```
Comment (vinnyjth) [inline packages/WillowCreekApp/src/index.js:100]: Nice work on this merge!

File: packages/WillowCreekApp/src/ui/Onboarding/index.js [hunk 3]
```diff
@@ -20,30 +26,37 @@ function Onboarding({ navigation }) {
           <>
             {/* <AskName onPressPrimary={swipeForward} /> */}
             {/* <AboutYouWithFirstName onPressPrimary={swipeForward} /> */}
-            <AskNotifications
```
Comment (vinnyjth) [inline packages/WillowCreekApp/src/ui/Onboarding/index.js:23]: Another nice merge :) These are always tricky.

File: packages/WillowCreekApp/src/LandingScreen.js [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
+import { Image, StyleSheet } from 'react-native';
+import { styled, ConnectedImage } from '@apollosproject/ui-kit';
 
 import ApollosLandingScreen from './ui/LandingScreen';
 
 const FullScreenImage = styled({
   resizeMode: 'cover',
-  position: 'absolute',
-})(GradientOverlayImage);
+  ...StyleSheet.absoluteFill,
```
Comment (vinnyjth) [inline packages/WillowCreekApp/src/LandingScreen.js:9]: 👍

File: packages/apollos-api/src/data/RockConstants.js (hunk 1 above)
Comment (christinamegura) [reply packages/apollos-api/src/data/RockConstants.js]: Not that I can recall. I think I saw some changes coming in from master and didn't accept them. I have corrected this and added the file back in

File: packages/WillowCreekApp/jest.setup.js [hunk 5]
```diff
@@ -22,6 +22,14 @@ jest.mock('react-navigation', () => {
...
 });
 
+jest.mock('@react-navigation/native', () => {
+  const ActualNavigation = require.requireActual('@react-navigation/native');
+  return {
+    ...ActualNavigation,
+    SafeAreaView: require.requireActual('SafeAreaView'),
+  };
+});
+
```
Comment (christinamegura) [inline packages/WillowCreekApp/jest.setup.js:32]: Can you help me understand the difference between this and the react-navigation mock (directly above)? I think my snaps aren't updating because of this and I'm having a hard time understanding the error.

Comment (christinamegura) [reply packages/WillowCreekApp/jest.setup.js:32]: @vinnyjth 😃

Comment (christinamegura) [reply packages/WillowCreekApp/jest.setup.js:32]: I added `@react-navigation/.*` to `transformIgnorePatterns` in `package.json jest` config which seems to have worked to resolve the error I was getting but now I'm getting

**Review by vinnyjth — APPROVED** [review]
Looks good @christinamegura! The upgrade looks great, I couldn't find any issues and your changes looked very well applied.

**Comment by vinnyjth** [comment]
One thing to note: 

```
Updates both vertical and horizontal card mapping by adding an override for the Component prop of the cardComponentMapper
```

This PR accomplishes this goal for the vertical cards, but not the horizontal cards. As of now, there is now way to override the CardMapper for the horizontal cards throughout the app however, so that's something we'll need to fix in core and then do in Willow.
